Cloudinary-pwa-kit/packages/pwa-cloudinary/app/components/cloudinary-widgets/index.jsx

```diff
@@ -5,15 +5,15 @@ import { cloudinary } from '../../../config/default'
 const RenderCloudinaryGalleryWidget = ({ cloudinaryImageGallery = {} }) => {
 
     useEffect(() => {
+        const gallery = document.querySelector('#cld-gallery');
+        while (gallery.firstChild) {
+            gallery.removeChild(gallery.firstChild)
+        }
         if (cloudinaryImageGallery && cloudinaryImageGallery.galleryEnabled) {
             const galleryOptions = cloudinaryImageGallery.cloudinaryImage.galleryWidget.options
             galleryOptions.queryParam = cloudinary.CLD_PGW_TRACKING_PARAM
             if (typeof window !== 'undefined' && window.cloudinary && window.cloudinary.galleryWidget) {
-                if (window.cldGallery) {
-                    cldGallery.update(galleryOptions)
-                } else {
-                    window.cldGallery = window.cloudinary.galleryWidget(galleryOptions)
-                }
+                const cldGallery = window.cloudinary.galleryWidget(galleryOptions)
```

Comment:
this will definitely cause a memory leak.

Please try 
if (window.cldGallery) {
                    gallery.style.visibility: hidden
                    cldGallery.update(galleryOptions)
                    gallery.style.visibility: visibile                   
                } else {
                    window.cldGallery = window.cloudinary.galleryWidget(galleryOptions)
                }

Reply:
I did the required changes, "style" is not available in cldGallery, so getting error on it.
Since we have removed the following code
`const gallery = document.querySelector('#cld-gallery')
        while (gallery.firstChild) {
            gallery.removeChild(gallery.firstChild)
        }`
 It seems its working fine by removing the above code

Reply:
then try 
const gallery = document.querySelector('#cld-gallery')
and then hide and unhide it.

Removing child nodes is prone to create issues, especially as we're reusing the widget through the customer's journey

Reply:
Did you deploy it?